common/src/chain/config.rs

```diff
@@ -147,7 +162,9 @@ const REGTEST_ADDRESS_PREFIX: &str = "rmt";
 // If block time is 2 minutes (which is my goal eventually), then 500 is equivalent to 100 in bitcoin's 10 minutes.
 const MAINNET_BLOCKREWARD_MATURITY: BlockDistance = BlockDistance::new(500);
 // DSA allows us to have blocks up to 1mb
-pub const MAX_BLOCK_WEIGHT: usize = 1_048_576;
+const MAX_BLOCK_HEADER_SIZE: usize = 1024;
+const MAX_BLOCK_TXS_SIZE: usize = 524_288;
+const MAX_BLOCK_CONSTRACTS_SIZE: usize = 524_288;
```

Comment:
Do we want this 50/50? I'm fine with this for now but I think we should revisit this and consider having more space for TXs.

Reply:
There is also the possibility of having total size (say 1MB) and a limit on contract space (0.5 MB for example). Implying TXs have 0.5MB reserved and could compete with contract stuff for the other 0.5MB so it could be up to full 1MB TXs given enough demand.